core/src/main/java/org/ehcache/events/CacheEventNotificationServiceImpl.java

```diff
@@ -82,9 +99,20 @@ public CacheEventNotificationServiceImpl(ExecutorService orderedDelivery, Execut
   public void registerCacheEventListener(CacheEventListener<? super K, ? super V> listener,
                                   EventOrdering ordering, EventFiring firing, EnumSet<EventType> forEventTypes) {
     boolean doRegister = forEventTypes.contains(EventType.EVICTED) || forEventTypes.contains(EventType.EXPIRED);
-    if (!registeredListeners.add(new EventListenerWrapper(listener, firing, ordering, forEventTypes))) {
+    EventListenerWrapper wrapper = new EventListenerWrapper(listener, firing, ordering, forEventTypes);
+    if(wrapper.config.orderingMode() == EventOrdering.ORDERED) {
+      orderedListenerCount.incrementAndGet();
+    }
+    
+    if(aSyncListenersSet.contains(wrapper) || syncListenersSet.contains(wrapper)) {
```

Comment:
No need to do that here, just observe the result of calling add below and throw when `false`